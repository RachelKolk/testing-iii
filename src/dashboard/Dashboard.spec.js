// Test away

import React from 'react';
import {render} from 'react-testing-library';
import 'jest-dom/extend-expect';
import 'react-testing-library/cleanup-after-each';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Dashboard from './Dashboard';

describe('<Dashboard />' , () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Dashboard />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

    it('matches snapshot', () => {
        const tree = renderer.create(<Dashboard />);
    
        expect(tree.toJSON()).toMatchSnapshot();
    });

    

});