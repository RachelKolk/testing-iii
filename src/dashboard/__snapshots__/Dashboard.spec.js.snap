// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="key"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="gate"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lockToggleButton"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="closeToggleButton"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
